svthalia/concrexit
# Add OpenAPI scheme generation and Swagger UI

### Summary

Add OpenAPI scheme generation and Swagger UI.

This uses Django Rest Framework's [built-in capability](https://www.django-rest-framework.org/api-guide/schemas/) to generate a [OpenAPI 3.0](https://swagger.io/specification/) specification of our website's API.

It also includes the latest version of Swagger UI, which has the ability to read the generated schema and use it directly from the browser. The PR also extends the basic built-in functionality to add oauth2 support to the docs, which makes it easy to test and try the API with OAuth scopes.

Most API endpoints are available, the event registration is missing an option for the information fields however. I'm afraid we can't do much about that currently.

### How to test
Steps to test the changes you made:
1. Go to `/api/docs`
2. Check that most endpoints are available
3. Try a couple of endpoints
